// @babel/types/lib/builders/generated/lowercase.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.anyTypeAnnotation = anyTypeAnnotation;
exports.argumentPlaceholder = argumentPlaceholder;
exports.arrayExpression = arrayExpression;
exports.arrayPattern = arrayPattern;
exports.arrayTypeAnnotation = arrayTypeAnnotation;
exports.arrowFunctionExpression = arrowFunctionExpression;
exports.assignmentExpression = assignmentExpression;
exports.assignmentPattern = assignmentPattern;
exports.awaitExpression = awaitExpression;
exports.bigIntLiteral = bigIntLiteral;
exports.binaryExpression = binaryExpression;
exports.bindExpression = bindExpression;
exports.blockStatement = blockStatement;
exports.booleanLiteral = booleanLiteral;
exports.booleanLiteralTypeAnnotation = booleanLiteralTypeAnnotation;
exports.booleanTypeAnnotation = booleanTypeAnnotation;
exports.breakStatement = breakStatement;
exports.callExpression = callExpression;
exports.catchClause = catchClause;
exports.classAccessorProperty = classAccessorProperty;
exports.classBody = classBody;
exports.classDeclaration = classDeclaration;
exports.classExpression = classExpression;
exports.classImplements = classImplements;
exports.classMethod = classMethod;
exports.classPrivateMethod = classPrivateMethod;
exports.classPrivateProperty = classPrivateProperty;
exports.classProperty = classProperty;
exports.conditionalExpression = conditionalExpression;
exports.continueStatement = continueStatement;
exports.debuggerStatement = debuggerStatement;
exports.decimalLiteral = decimalLiteral;
exports.declareClass = declareClass;
exports.declareExportAllDeclaration = declareExportAllDeclaration;
exports.declareExportDeclaration = declareExportDeclaration;
exports.declareFunction = declareFunction;
exports.declareInterface = declareInterface;
exports.declareModule = declareModule;
exports.declareModuleExports = declareModuleExports;
exports.declareOpaqueType = declareOpaqueType;
exports.declareTypeAlias = declareTypeAlias;
exports.declareVariable = declareVariable;
exports.declaredPredicate = declaredPredicate;
exports.decorator = decorator;
exports.directive = directive;
exports.directiveLiteral = directiveLiteral;
exports.doExpression = doExpression;
exports.doWhileStatement = doWhileStatement;
exports.emptyStatement = emptyStatement;
exports.emptyTypeAnnotation = emptyTypeAnnotation;
exports.enumBooleanBody = enumBooleanBody;
exports.enumBooleanMember = enumBooleanMember;
exports.enumDeclaration = enumDeclaration;
exports.enumDefaultedMember = enumDefaultedMember;
exports.enumNumberBody = enumNumberBody;
exports.enumNumberMember = enumNumberMember;
exports.enumStringBody = enumStringBody;
exports.enumStringMember = enumStringMember;
exports.enumSymbolBody = enumSymbolBody;
exports.existsTypeAnnotation = existsTypeAnnotation;
exports.exportAllDeclaration = exportAllDeclaration;
exports.exportDefaultDeclaration = exportDefaultDeclaration;
exports.exportDefaultSpecifier = exportDefaultSpecifier;
exports.exportNamedDeclaration = exportNamedDeclaration;
exports.exportNamespaceSpecifier = exportNamespaceSpecifier;
exports.exportSpecifier = exportSpecifier;
exports.expressionStatement = expressionStatement;
exports.file = file;
exports.forInStatement = forInStatement;
exports.forOfStatement = forOfStatement;
exports.forStatement = forStatement;
exports.functionDeclaration = functionDeclaration;
exports.functionExpression = functionExpression;
exports.functionTypeAnnotation = functionTypeAnnotation;
exports.functionTypeParam = functionTypeParam;
exports.genericTypeAnnotation = genericTypeAnnotation;
exports.identifier = identifier;
exports.ifStatement = ifStatement;
exports.import = _import;
exports.importAttribute = importAttribute;
exports.importDeclaration = importDeclaration;
exports.importDefaultSpecifier = importDefaultSpecifier;
exports.importExpression = importExpression;
exports.importNamespaceSpecifier = importNamespaceSpecifier;
exports.importSpecifier = importSpecifier;
exports.indexedAccessType = indexedAccessType;
exports.inferredPredicate = inferredPredicate;
exports.interfaceDeclaration = interfaceDeclaration;
exports.interfaceExtends = interfaceExtends;
exports.interfaceTypeAnnotation = interfaceTypeAnnotation;
exports.interpreterDirective = interpreterDirective;
exports.intersectionTypeAnnotation = intersectionTypeAnnotation;
exports.jSXAttribute = exports.jsxAttribute = jsxAttribute;
exports.jSXClosingElement = exports.jsxClosingElement = jsxClosingElement;
exports.jSXClosingFragment = exports.jsxClosingFragment = jsxClosingFragment;
exports.jSXElement = exports.jsxElement = jsxElement;
exports.jSXEmptyExpression = exports.jsxEmptyExpression = jsxEmptyExpression;
exports.jSXExpressionContainer = exports.jsxExpressionContainer = jsxExpressionContainer;
exports.jSXFragment = exports.jsxFragment = jsxFragment;
exports.jSXIdentifier = exports.jsxIdentifier = jsxIdentifier;
exports.jSXMemberExpression = exports.jsxMemberExpression = jsxMemberExpression;
exports.jSXNamespacedName = exports.jsxNamespacedName = jsxNamespacedName;
exports.jSXOpeningElement = exports.jsxOpeningElement = jsxOpeningElement;
exports.jSXOpeningFragment = exports.jsxOpeningFragment = jsxOpeningFragment;
exports.jSXSpreadAttribute = exports.jsxSpreadAttribute = jsxSpreadAttribute;
exports.jSXSpreadChild = exports.jsxSpreadChild = jsxSpreadChild;
exports.jSXText = exports.jsxText = jsxText;
exports.labeledStatement = labeledStatement;
exports.logicalExpression = logicalExpression;
exports.memberExpression = memberExpression;
exports.metaProperty = metaProperty;
exports.mixedTypeAnnotation = mixedTypeAnnotation;
exports.moduleExpression = moduleExpression;
exports.newExpression = newExpression;
exports.noop = noop;
exports.nullLiteral = nullLiteral;
exports.nullLiteralTypeAnnotation = nullLiteralTypeAnnotation;
exports.nullableTypeAnnotation = nullableTypeAnnotation;
exports.numberLiteral = NumberLiteral;
exports.numberLiteralTypeAnnotation = numberLiteralTypeAnnotation;
exports.numberTypeAnnotation = numberTypeAnnotation;
exports.numericLiteral = numericLiteral;
exports.objectExpression = objectExpression;
exports.objectMethod = objectMethod;
exports.objectPattern = objectPattern;
exports.objectProperty = objectProperty;
exports.objectTypeAnnotation = objectTypeAnnotation;
exports.objectTypeCallProperty = objectTypeCallProperty;
exports.objectTypeIndexer = objectTypeIndexer;
exports.objectTypeInternalSlot = objectTypeInternalSlot;
exports.objectTypeProperty = objectTypeProperty;
exports.objectTypeSpreadProperty = objectTypeSpreadProperty;
exports.opaqueType = opaqueType;
exports.optionalCallExpression = optionalCallExpression;
exports.optionalIndexedAccessType = optionalIndexedAccessType;
exports.optionalMemberExpression = optionalMemberExpression;
exports.parenthesizedExpression = parenthesizedExpression;
exports.pipelineBareFunction = pipelineBareFunction;
exports.pipelinePrimaryTopicReference = pipelinePrimaryTopicReference;
exports.pipelineTopicExpression = pipelineTopicExpression;
exports.placeholder = placeholder;
exports.privateName = privateName;
exports.program = program;
exports.qualifiedTypeIdentifier = qualifiedTypeIdentifier;
exports.recordExpression = recordExpression;
exports.regExpLiteral = regExpLiteral;
exports.regexLiteral = RegexLiteral;
exports.restElement = restElement;
exports.restProperty = RestProperty;
exports.returnStatement = returnStatement;
exports.sequenceExpression = sequenceExpression;
exports.spreadElement = spreadElement;
exports.spreadProperty = SpreadProperty;
exports.staticBlock = staticBlock;
exports.stringLiteral = stringLiteral;
exports.stringLiteralTypeAnnotation = stringLiteralTypeAnnotation;
exports.stringTypeAnnotation = stringTypeAnnotation;
exports.super = _super;
exports.switchCase = switchCase;
exports.switchStatement = switchStatement;
exports.symbolTypeAnnotation = symbolTypeAnnotation;
exports.taggedTemplateExpression = taggedTemplateExpression;
exports.templateElement = templateElement;
exports.templateLiteral = templateLiteral;
exports.thisExpression = thisExpression;
exports.thisTypeAnnotation = thisTypeAnnotation;
exports.throwStatement = throwStatement;
exports.topicReference = topicReference;
exports.tryStatement = tryStatement;
exports.tSAnyKeyword = exports.tsAnyKeyword = tsAnyKeyword;
exports.tSArrayType = exports.tsArrayType = tsArrayType;
exports.tSAsExpression = exports.tsAsExpression = tsAsExpression;
exports.tSBigIntKeyword = exports.tsBigIntKeyword = tsBigIntKeyword;
exports.tSBooleanKeyword = exports.tsBooleanKeyword = tsBooleanKeyword;
exports.tSCallSignatureDeclaration = exports.tsCallSignatureDeclaration = tsCallSignatureDeclaration;
exports.tSConditionalType = exports.tsConditionalType = tsConditionalType;
exports.tSConstructSignatureDeclaration = exports.tsConstructSignatureDeclaration = tsConstructSignatureDeclaration;
exports.tSConstructorType = exports.tsConstructorType = tsConstructorType;
exports.tSDeclareFunction = exports.tsDeclareFunction = tsDeclareFunction;
exports.tSDeclareMethod = exports.tsDeclareMethod = tsDeclareMethod;
exports.tSEnumBody = exports.tsEnumBody = tsEnumBody;
exports.tSEnumDeclaration = exports.tsEnumDeclaration = tsEnumDeclaration;
exports.tSEnumMember = exports.tsEnumMember = tsEnumMember;
exports.tSExportAssignment = exports.tsExportAssignment = tsExportAssignment;
exports.tSExpressionWithTypeArguments = exports.tsExpressionWithTypeArguments = tsExpressionWithTypeArguments;
exports.tSExternalModuleReference = exports.tsExternalModuleReference = tsExternalModuleReference;
exports.tSFunctionType = exports.tsFunctionType = tsFunctionType;
exports.tSImportEqualsDeclaration = exports.tsImportEqualsDeclaration = tsImportEqualsDeclaration;
exports.tSImportType = exports.tsImportType = tsImportType;
exports.tSIndexSignature = exports.tsIndexSignature = tsIndexSignature;
exports.tSIndexedAccessType = exports.tsIndexedAccessType = tsIndexedAccessType;
exports.tSInferType = exports.tsInferType = tsInferType;
exports.tSInstantiationExpression = exports.tsInstantiationExpression = tsInstantiationExpression;
exports.tSInterfaceBody = exports.tsInterfaceBody = tsInterfaceBody;
exports.tSInterfaceDeclaration = exports.tsInterfaceDeclaration = tsInterfaceDeclaration;
exports.tSIntersectionType = exports.tsIntersectionType = tsIntersectionType;
exports.tSIntrinsicKeyword = exports.tsIntrinsicKeyword = tsIntrinsicKeyword;
exports.tSLiteralType = exports.tsLiteralType = tsLiteralType;
exports.tSMappedType = exports.tsMappedType = tsMappedType;
exports.tSMethodSignature = exports.tsMethodSignature = tsMethodSignature;
exports.tSModuleBlock = exports.tsModuleBlock = tsModuleBlock;
exports.tSModuleDeclaration = exports.tsModuleDeclaration = tsModuleDeclaration;
exports.tSNamedTupleMember = exports.tsNamedTupleMember = tsNamedTupleMember;
exports.tSNamespaceExportDeclaration = exports.tsNamespaceExportDeclaration = tsNamespaceExportDeclaration;
exports.tSNeverKeyword = exports.tsNeverKeyword = tsNeverKeyword;
exports.tSNonNullExpression = exports.tsNonNullExpression = tsNonNullExpression;
exports.tSNullKeyword = exports.tsNullKeyword = tsNullKeyword;
exports.tSNumberKeyword = exports.tsNumberKeyword = tsNumberKeyword;
exports.tSObjectKeyword = exports.tsObjectKeyword = tsObjectKeyword;
exports.tSOptionalType = exports.tsOptionalType = tsOptionalType;
exports.tSParameterProperty = exports.tsParameterProperty = tsParameterProperty;
exports.tSParenthesizedType = exports.tsParenthesizedType = tsParenthesizedType;
exports.tSPropertySignature = exports.tsPropertySignature = tsPropertySignature;
exports.tSQualifiedName = exports.tsQualifiedName = tsQualifiedName;
exports.tSRestType = exports.tsRestType = tsRestType;
exports.tSSatisfiesExpression = exports.tsSatisfiesExpression = tsSatisfiesExpression;
exports.tSStringKeyword = exports.tsStringKeyword = tsStringKeyword;
exports.tSSymbolKeyword = exports.tsSymbolKeyword = tsSymbolKeyword;
exports.tSTemplateLiteralType = exports.tsTemplateLiteralType = tsTemplateLiteralType;
exports.tSThisType = exports.tsThisType = tsThisType;
exports.tSTupleType = exports.tsTupleType = tsTupleType;
exports.tSTypeAliasDeclaration = exports.tsTypeAliasDeclaration = tsTypeAliasDeclaration;
exports.tSTypeAnnotation = exports.tsTypeAnnotation = tsTypeAnnotation;
exports.tSTypeAssertion = exports.tsTypeAssertion = tsTypeAssertion;
exports.tSTypeLiteral = exports.tsTypeLiteral = tsTypeLiteral;
exports.tSTypeOperator = exports.tsTypeOperator = tsTypeOperator;
exports.tSTypeParameter = exports.tsTypeParameter = tsTypeParameter;
exports.tSTypeParameterDeclaration = exports.tsTypeParameterDeclaration = tsTypeParameterDeclaration;
exports.tSTypeParameterInstantiation = exports.tsTypeParameterInstantiation = tsTypeParameterInstantiation;
exports.tSTypePredicate = exports.tsTypePredicate = tsTypePredicate;
exports.tSTypeQuery = exports.tsTypeQuery = tsTypeQuery;
exports.tSTypeReference = exports.tsTypeReference = tsTypeReference;
exports.tSUndefinedKeyword = exports.tsUndefinedKeyword = tsUndefinedKeyword;
exports.tSUnionType = exports.tsUnionType = tsUnionType;
exports.tSUnknownKeyword = exports.tsUnknownKeyword = tsUnknownKeyword;
exports.tSVoidKeyword = exports.tsVoidKeyword = tsVoidKeyword;
exports.tupleExpression = tupleExpression;
exports.tupleTypeAnnotation = tupleTypeAnnotation;
exports.typeAlias = typeAlias;
exports.typeAnnotation = typeAnnotation;
exports.typeCastExpression = typeCastExpression;
exports.typeParameter = typeParameter;
exports.typeParameterDeclaration = typeParameterDeclaration;
exports.typeParameterInstantiation = typeParameterInstantiation;
exports.typeofTypeAnnotation = typeofTypeAnnotation;
exports.unaryExpression = unaryExpression;
exports.unionTypeAnnotation = unionTypeAnnotation;
exports.updateExpression = updateExpression;
exports.v8IntrinsicIdentifier = v8IntrinsicIdentifier;
exports.variableDeclaration = variableDeclaration;
exports.variableDeclarator = variableDeclarator;
exports.variance = variance;
exports.voidPattern = voidPattern;
exports.voidTypeAnnotation = voidTypeAnnotation;
exports.whileStatement = whileStatement;
exports.withStatement = withStatement;
exports.yieldExpression = yieldExpression;
var _validate = require("../../validators/validate.js");
var _deprecationWarning = require("../../utils/deprecationWarning.js");
var utils = require("../../definitions/utils.js");
const {
  validateInternal: validate
} = _validate;
const {
  NODE_FIELDS
} = utils;
function bigIntLiteral(value) {
  if (typeof value === "bigint") {
    value = value.toString();
  }
  const node = {
    type: "BigIntLiteral",
    value
  };
  const defs = NODE_FIELDS.BigIntLiteral;
  validate(defs.value, node, "value", value);
  return node;
}
function arrayExpression(elements = []) {
  const node = {
    type: "ArrayExpression",
    elements
  };
  const defs = NODE_FIELDS.ArrayExpression;
  validate(defs.elements, node, "elements", elements, 1);
  return node;
}
function assignmentExpression(operator, left, right) {
  const node = {
    type: "AssignmentExpression",
    operator,
    left,
    right
  };
  const defs = NODE_FIELDS.AssignmentExpression;
  validate(defs.operator, node, "operator", operator);
  validate(defs.left, node, "left", left, 1);
  validate(defs.right, node, "right", right, 1);
  return node;
}
function binaryExpression(operator, left, right) {
  const node = {
    type: "BinaryExpression",
    operator,
    left,
    right
  };
  const defs = NODE_FIELDS.BinaryExpression;
  validate(defs.operator, node, "operator", operator);
  validate(defs.left, node, "left", left, 1);
  validate(defs.right, node, "right", right, 1);
  return node;
}
function interpreterDirective(value) {
  const node = {
    type: "InterpreterDirective",
    value
  };
  const defs = NODE_FIELDS.InterpreterDirective;
  validate(defs.value, node, "value", value);
  return node;
}
function directive(value) {
  const node = {
    type: "Directive",
    value
  };
  const defs = NODE_FIELDS.Directive;
  validate(defs.value, node, "value", value, 1);
  return node;
}
function directiveLiteral(value) {
  const node = {
    type: "DirectiveLiteral",
    value
  };
  const defs = NODE_FIELDS.DirectiveLiteral;
  validate(defs.value, node, "value", value);
  return node;
}
function blockStatement(body, directives = []) {
  const node = {
    type: "BlockStatement",
    body,
    directives
  };
  const defs = NODE_FIELDS.BlockStatement;
  validate(defs.body, node, "body", body, 1);
  validate(defs.directives, node, "directives", directives, 1);
  return node;
}
function breakStatement(label = null) {
  const node = {
    type: "BreakStatement",
    label
  };
  const defs = NODE_FIELDS.BreakStatement;
  validate(defs.label, node, "label", label, 1);
  return node;
}
function callExpression(callee, _arguments) {
  const node = {
    type: "CallExpression",
    callee,
    arguments: _arguments
  };
  const defs = NODE_FIELDS.CallExpression;
  validate(defs.callee, node, "callee", callee, 1);
  validate(defs.arguments, node, "arguments", _arguments, 1);
  return node;
}
function catchClause(param = null, body) {
  const node = {
    type: "CatchClause",
    param,
    body
  };
  const defs = NODE_FIELDS.CatchClause;
  validate(defs.param, node, "param", param, 1);
  validate(defs.body, node, "body", body, 1);
  return node;
}
function conditionalExpression(test, consequent, alternate) {
  const node = {
    type: "ConditionalExpression",
    test,
    consequent,
    alternate
  };
  const defs = NODE_FIELDS.ConditionalExpression;
  validate(defs.test, node, "test", test, 1);
  validate(defs.consequent, node, "consequent", consequent, 1);
  validate(defs.alternate, node, "alternate", alternate, 1);
  return node;
}
function continueStatement(label = null) {
  const node = {
    type: "ContinueStatement",
    label
  };
  const defs = NODE_FIELDS.ContinueStatement;
  validate(defs.label, node, "label", label, 1);
  return node;
}
function debuggerStatement() {
  return {
    type: "DebuggerStatement"
  };
}
function doWhileStatement(test, body) {
  const node = {
    type: "DoWhileStatement",
    test,
    body
  };
  const defs = NODE_FIELDS.DoWhileStatement;
  validate(defs.test, node, "test", test, 1);
  validate(defs.body, node, "body", body, 1);
  return node;
}
function emptyStatement() {
  return {
    type: "EmptyStatement"
  };
}
function expressionStatement(expression) {
  const node = {
    type: "ExpressionStatement",
    expression
  };
  const defs = NODE_FIELDS.ExpressionStatement;
  validate(defs.expression, node, "expression", expression, 1);
  return node;
}
function file(program, comments = null, tokens = null) {
  const node = {
    type: "File",
    program,
    comments,
    tokens
  };
  const defs = NODE_FIELDS.File;
  validate(defs.program, node, "program", program, 1);
  validate(defs.comments, node, "comments", comments, 1);
  validate(defs.tokens, node, "tokens", tokens);
  return node;
}
function forInStatement(left, right, body) {
  const node = {
    type: "ForInStatement",
    left,
    right,
    body
  };
  const defs = NODE_FIELDS.ForInStatement;
  validate(defs.left, node, "left", left, 1);
  validate(defs.right, node, "right", right, 1);
  validate(defs.body, node, "body", body, 1);
  return node;
}
function forStatement(init = null, test = null, update = null, body) {
  const node = {
    type: "ForStatement",
    init,
    test,
    update,
    body
  };
  const defs = NODE_FIELDS.ForStatement;
  validate(defs.init, node, "init", init, 1);
  validate(defs.test, node, "test", test, 1);
  validate(defs.update, node, "update", update, 1);
  validate(defs.body, node, "body", body, 1);
  return node;
}
function functionDeclaration(id = null, params, body, generator = false, async = false) {
  const node = {
    type: "FunctionDeclaration",
    id,
    params,
    body,
    generator,
    async
  };
  const defs = NODE_FIELDS.FunctionDeclaration;
  validate(defs.id, node, "id", id, 1);
  validate(defs.params, node, "params", params, 1);
  validate(defs.body, node, "body", body, 1);
  validate(defs.generator, node, "generator", generator);
  validate(defs.async, node, "async", async);
  return node;
}
function functionExpression(id = null, params, body, generator = false, async = false) {
  const node = {
    type: "FunctionExpression",
    id,
    params,
    body,
    generator,
    async
  };
  const defs = NODE_FIELDS.FunctionExpression;
  validate(defs.id, node, "id", id, 1);
  validate(defs.params, node, "params", params, 1);
  validate(defs.body, node, "body", body, 1);
  validate(defs.generator, node, "generator", generator);
  validate(defs.async, node, "async", async);
  return node;
}
function identifier(name) {
  const node = {
    type: "Identifier",
    name
  };
  const defs = NODE_FIELDS.Identifier;
  validate(defs.name, node, "name", name);
  return node;
}
function ifStatement(test, consequent, alternate = null) {
  const node = {
    type: "IfStatement",
    test,
    consequent,
    alternate
  };
  const defs = NODE_FIELDS.IfStatement;
  validate(defs.test, node, "test", test, 1);
  validate(defs.consequent, node, "consequent", consequent, 1);
  validate(defs.alternate, node, "alternate", alternate, 1);
  return node;
}
function labeledStatement(label, body) {
  const node = {
    type: "LabeledStatement",
    label,
    body
  };
  const defs = NODE_FIELDS.LabeledStatement;
  validate(defs.label, node, "label", label, 1);
  validate(defs.body, node, "body", body, 1);
  return node;
}
function stringLiteral(value) {
  const node = {
    type: "StringLiteral",
    value
  };
  const defs = NODE_FIELDS.StringLiteral;
  validate(defs.value, node, "value", value);
  return node;
}
function numericLiteral(value) {
  const node = {
    type: "NumericLiteral",
    value
  };
  const defs = NODE_FIELDS.NumericLiteral;
  validate(defs.value, node, "value", value);
  return node;
}
function nullLiteral() {
  return {
    type: "NullLiteral"
  };
}
function booleanLiteral(value) {
  const node = {
    type: "BooleanLiteral",
    value
  };
  const defs = NODE_FIELDS.BooleanLiteral;
  validate(defs.value, node, "value", value);
  return node;
}
function regExpLiteral(pattern, flags = "") {
  const node = {
    type: "RegExpLiteral",
    pattern,
    flags
  };
  const defs = NODE_FIELDS.RegExpLiteral;
  validate(defs.pattern, node, "pattern", pattern);
  validate(defs.flags, node, "flags", flags);
  return node;
}
function logicalExpression(operator, left, right) {
  const node = {
    type: "LogicalExpression",
    operator,
    left,
    right
  };
  const defs = NODE_FIELDS.LogicalExpression;
  validate(defs.operator, node, "operator", operator);
  validate(defs.left, node, "left", left, 1);
  validate(defs.right, node, "right", right, 1);
  return node;
}
function memberExpression(object, property, computed = false, optional = null) {
  const node = {
    type: "MemberExpression",
    object,
    property,
    computed,
    optional
  };
  const defs = NODE_FIELDS.MemberExpression;
  validate(defs.object, node, "object", object, 1);
  validate(defs.property, node, "property", property, 1);
  validate(defs.computed, node, "computed", computed);
  validate(defs.optional, node, "optional", optional);
  return node;
}
function newExpression(callee, _arguments) {
  const node = {
    type: "NewExpression",
    callee,
    arguments: _arguments
  };
  const defs = NODE_FIELDS.NewExpression;
  validate(defs.callee, node, "callee", callee, 1);
  validate(defs.arguments, node, "arguments", _arguments, 1);
  return node;
}
function program(body, directives = [], sourceType = "script", interpreter = null) {
  const node = {
    type: "Program",
    body,
    directives,
    sourceType,
    interpreter
  };
  const defs = NODE_FIELDS.Program;
  validate(defs.body, node, "body", body, 1);
  validate(defs.directives, node, "directives", directives, 1);
  validate(defs.sourceType, node, "sourceType", sourceType);
  validate(defs.interpreter, node, "interpreter", interpreter, 1);
  return node;
}
function objectExpression(properties) {
  const node = {
    type: "ObjectExpression",
    properties
  };
  const defs = NODE_FIELDS.ObjectExpression;
  validate(defs.properties, node, "properties", properties, 1);
  return node;
}
function objectMethod(kind = "method", key, params, body, computed = false, generator = false, async = false) {
  const node = {
    type: "ObjectMethod",
    kind,
    key,
    params,
    body,
    computed,
    generator,
    async
  };
  const defs = NODE_FIELDS.ObjectMethod;
  validate(defs.kind, node, "kind", kind);
  validate(defs.key, node, "key", key, 1);
  validate(defs.params, node, "params", params, 1);
  validate(defs.body, node, "body", body, 1);
  validate(defs.computed, node, "computed", computed);
  validate(defs.generator, node, "generator", generator);
  validate(defs.async, node, "async", async);
  return node;
}
function objectProperty(key, value, computed = false, shorthand = false, decorators = null) {
  const node = {
    type: "ObjectProperty",
    key,
    value,
    computed,
    shorthand,
    decorators
  };
  const defs = NODE_FIELDS.ObjectProperty;
  validate(defs.key, node, "key", key, 1);
  validate(defs.value, node, "value", value, 1);
  validate(defs.computed, node, "computed", computed);
  validate(defs.shorthand, node, "shorthand", shorthand);
  validate(defs.decorators, node, "decorators", decorators, 1);
  return node;
}
function restElement(argument) {
  const node = {
    type: "RestElement",
    argument
  };
  const defs = NODE_FIELDS.RestElement;
  validate(defs.argument, node, "argument", argument, 1);
  return node;
}
function returnStatement(argument = null) {
  const node = {
    type: "ReturnStatement",
    argument
  };
  const defs = NODE_FIELDS.ReturnStatement;
  validate(defs.argument, node, "argument", argument, 1);
  return node;
}
function sequenceExpression(expressions) {
  const node = {
    type: "SequenceExpression",
    expressions
  };
  const defs = NODE_FIELDS.SequenceExpression;
  validate(defs.expressions, node, "expressions", expressions, 1);
  return node;
}
function parenthesizedExpression(expression) {
  const node = {
    type: "ParenthesizedExpression",
    expression
  };
  const defs = NODE_FIELDS.ParenthesizedExpression;
  validate(defs.expression, node, "expression", expression, 1);
  return node;
}
function switchCase(test = null, consequent) {
  const node = {
    type: "SwitchCase",
    test,
    consequent
  };
  const defs = NODE_FIELDS.SwitchCase;
  validate(defs.test, node, "test", test, 1);
  validate(defs.consequent, node, "consequent", consequent, 1);
  return node;
}
function switchStatement(discriminant, cases) {
  const node = {
    type: "SwitchStatement",
    discriminant,
    cases
  };
  const defs = NODE_FIELDS.SwitchStatement;
  validate(defs.discriminant, node, "discriminant", discriminant, 1);
  validate(defs.cases, node, "cases", cases, 1);
  return node;
}
function thisExpression() {
  return {
    type: "ThisExpression"
  };
}
function throwStatement(argument) {
  const node = {
    type: "ThrowStatement",
    argument
  };
  const defs = NODE_FIELDS.ThrowStatement;
  validate(defs.argument, node, "argument", argument, 1);
  return node;
}
function tryStatement(block, handler = null, finalizer = null) {
  const node = {
    type: "TryStatement",
    block,
    handler,
    finalizer
  };
  const defs = NODE_FIELDS.TryStatement;
  validate(defs.block, node, "block", block, 1);
  validate(defs.handler, node, "handler", handler, 1);
  validate(defs.finalizer, node, "finalizer", finalizer, 1);
  return node;
}
function unaryExpression(operator, argument, prefix = true) {
  const node = {
    type: "UnaryExpression",
    operator,
    argument,
    prefix
  };
  const defs = NODE_FIELDS.UnaryExpression;
  validate(defs.operator, node, "operator", operator);
  validate(defs.argument, node, "argument", argument, 1);
  validate(defs.prefix, node, "prefix", prefix);
  return node;
}
function updateExpression(operator, argument, prefix = false) {
  const node = {
    type: "UpdateExpression",
    operator,
    argument,
    prefix
  };
  const defs = NODE_FIELDS.UpdateExpression;
  validate(defs.operator, node, "operator", operator);
  validate(defs.argument, node, "argument", argument, 1);
  validate(defs.prefix, node, "prefix", prefix);
  return node;
}
function variableDeclaration(kind, declarations) {
  const node = {
    type: "VariableDeclaration",
    kind,
    declarations
  };
  const defs = NODE_FIELDS.VariableDeclaration;
  validate(defs.kind, node, "kind", kind);
  validate(d6-5cS2piĬi`y&3N|>xM01眔\hK&ӕSDvBax~	9پA^?:ȲLC*EegD#gn1ld̀mUM@zdg|I6F#y.pUuteh`zlkA[&
jD#N1F{>y&Ct]ud:/m& EMK*ٿ@ʇMC{Ɉ= gzrg`t3SMpZ}4k/:@rPu9H,t1烁kSEKsB
"73BM\%wx@qFѬW'}5@6Y{\e9cQw>0O:9/[߫Z~ʠxl2GQd;q$%:;AM:[R$^03Nb3,ӷ%3e9X):53ȇQAц6hOa7kr-gGKSɗt7){:.nvd@3+&z%R; S=
3>@G##ȋP)
,P,XR2+ǷW UOwЌ1%7睯(3w`'3 (=^e(JrbEYmf	\v-%?5leӑ;CqC߇X'"
v^/)FHt</}ܑ
AF}J
ā^THd֮Jf̹%exW8ԐUrvNhkZц[nڨ. ՙR'Ҝ/KzʄĎkA f1m1yݝ=yF˙
v&`0L4U)㗴EoXyB
otj?5hM4@4VuG\At8ŗ&+]x`oFP
O\@/jתl\# xV
j4V]Ranh428;p\L D`uآ.o}Y8'?*z;e'6L`Tᴗ>mC@|y0l痿ҢKc t-k!0ĩηd9p[;２#p/LuSIȾ3sůw6|L`˩5=6l#PIc1%CI`8J];ZI9X8jҢ_9ȫFO9[m;.Lv  #\d_zmt=_SN^׫
D+7Ӽf2w~D\z@
2P鞊N56L?q8Ӄ:C?rt{&ᒗ򷾈H~Ѥ{ޡI,r\ǫĦ:\٦Y@ؔلtW{+?JVLkyaqrQ$G^	ezO&X#K~,$Qi'1X6Ycui,
	.`,g>V
e^r	Gb%ً	R[W0Gk822YCg'd?]f Wug<εp}
G<r2Q3KLu+:Lj+>
. T`)l0r.w{ͼSot`c9Dk
%Jǐұ)/_ACV3OuUS%CvMyﶴ_idRnPSerKY+5ykTgF`jzNqǋ84bjHxs%fXcJ0TJ	j2U@S/ V/Vd!x9އt<&,T$hn$K	h@2A "ofZ ܧOo)"yaхІo]?IF'eq*g^{yPWHOOK9eFLhvRss~<4ٮ
5F8Cň .J_cر#-c$sG"МRcA3hE)aP@2w9nC'`|<8TSsm߫ᅃT%>,?iV
HԐi Ko6^K'm+JJ&$CL:
>8ѭ"-+)/kWAqWjw<S@A=7$G9AhVtL0
>liTÄ,
Q"ƒN'|FH'1̾*k9< yvuz9d,"'HP8*FY? *{CWnCPB9IIcb:6K(hZ+_x^Ե׿Ā#-ٽvtc_Bi̡Fֶ6?}]8w'uE`3q""}=?@}@u<AZg)n1zy0̕u>eX	MS$gQr4]KČ%qӋvvl}IНAǿs1͞]P@D
`+4^ۑğ?dMejU2`x%RğU~I¦qo	{> fm]KNq"$I	cV:fGWݕ8ZJ66鉞F~MVd3Ccs؍'h>h'?+_Fy3rO+[):g#TfOS]b5 h!mF}}cy	`hH@=ټm*;,nUś)k^9Ȼ^TCJô7p}<TIx@!u]BJ^ tb	5O]XVI6 3r4L-"/`.P6y褘0,%[%&34srqD_Sxo;XntR*
9`[^V0ZzTEInQw@**K7-s8VxU\ҥiT1Z[kE؂^4~t< aԻWޙ;GS+¦:Ŕd=Sw(-KcMhID)gޔ#x5X+lA\S%;Y;!<j3ߠf3A-5ű*ά݈._),~dr;Fj}ֵVwi{E +f_20(m<h6'=.NFvUEq]Hz{jy_Fj0rh#Q+lQ͛`ɛ;an)mS+\Ko D5[6	lYk}>5r_j!ڙ/5}HAW
!_llب.f,9MVz.E3iFx'K5Fxl<HQl{@1M|ϲŲ3#{Ki:3 ع$9E79 X~Tlzl"TN
fz~k Xf$XOUEipjo
^ϚtdU[yt?V1hTܮ*#2t]"[y>+oY>z$W¶ÀniV8>"l!G΄q?ٴ@e*BY5|>2E!tbLXWr[քOC;(D@z͘OZ,JE71I3wܩ-+K[5]`xKpM?`ե}9-;6G"+Kwд'C,
#D}
:)]Q*@KfYVn P1TԿ^va+@bN]ڻ@
U5VVcs\M]	.{7ÜC>w~B#JY tޥ#Z'u4sjkDUP1F&J`םÓuh?7V0u\^xڔƹzx12+mxlXd,bwyIW1<)+>r#Ada0a݈zE
?FDf6l NRI#m]_<DgHoGBtZMsa:oW+rla+׍5 ɪq)ádMYSӂZ
1Xqgӝݹ1^<̴W7W:\6c\%bP፱h&_%1Z0Ԩxo/LiKH3hKz9)ݡts}6;9ѣxB_)38¥EBsq=<NT2Q>b`'*O%j]$L1ɯ)@7ƭyPfъxbUmw<5:X`q:_SL'N(S=1%(gt:DYC-^-d)(p@
%z<,0a5 v4d)!Ubi{{uޞ5οAԭM$
oʔ
ޟ&.MbTn|mά
ݷ@clTI;%OPfa;&c6 ?re/l
մHBm:i$ЉtUˮ_unEܧpg}jV۷ȴ	慲)$L0 mq@"|K\;*<)=텃r/	u@8E<ۀ8"7p}wnd fQZ0F`bjHzܰ ձ'%TqfohJ8̼}C{xI3%BIi4eYPbpx)R!1 	lYNSڠ8aH/ap{bj;Ld+י2P SD%.K5#=~qa+9szq3[){m8b;̵JS:|aWkzڒ>[9a"IesK%}Mw=dbWOG	
e~WccfuVXy)\f.[Gdk6#^d1!2k.͟w~:"VN*1*#oI*12XhjS̚yvCUH].uɎȪ/"mdGމcgD0B2,yd BcCwI]8-ǌ؞<q*EHpǲtH;w󈑧/5!K(;>~'R2LX?;@@̸a::*оؘXoGt7knFM8eZ`oS0ϪKaʔB'Po$vgDBY-Xo#|6 .FP"]!nSPtl}ƫk;^mfM'p]cK&hF\J
vkK_agK؄B 
ao;ίw nr[qj~V%o4Г5.!2;X[?/JTN\~(p?#R䛔uqldэ
J~5Tʶ3;B㣺sjak8|${iLAhjlS'[i$vypn_Yb$ H)įȕN[&szn9	Fzrz$u
RFoz>.`}~qwnOj4KZZѫ	W e)(#}ui6ҽHklUG9غTQiƩ4	cC0 FӅ41%<A}iaؒv8p|d$Y}*V!O2$&&&@q@^
]6
*-0N?iwG(y\uX0o5U=yRG[g"o	vюpH~cH>n:-_;9z_9쒟6}rYѤ?jޡ%8pڌsD6=M)'[ ŏ&ʬK&x. @HI..uŢ s ՚6_E8?Ȱg$p[6^]nit
Sun%yG6"Dvgl``#)%z{%wZMHB
ӨʽTSӧ.$ӲRQ"rpqoQ gsٌFMح,|y |'?Uh<؄T;5DoYy	/q;&h)Mǰuv,ƃ'nX~Gs[jNnL,TM'k`%YȹqM\mCQkIʫ9Χ.lIp_
1\zmy.XdZq{A!n0W y罒EʼR{jw[Дbc$mӞf2j] `&!+\=.e+I}il`3bRҍۙvHdAQ
8b=P],:rn6j)J+_vȧ+զh#![6O):2pmxL;y#Yz*GPaH:;h ,ay˅sYZ*-c'yajB0UQx8J
Yfd8t
kki,ZYCǂ&$z%P&;r|9GSWcOm&cZm6_vD*dC]EXN:xJ)76rr9G1r32(=yȊO`CdvGɏQběUV3ڡd@fSU3O ftm>6zËK$lw,h[s$\D%֏_cJ݁邖-PĹV93y*y)'Y"+#O0m?%<L?K %NNK)I>t7hoU4ހM 08.Q68w`9{,W^7`)Nڡl()G1pTYW)I>?\-3Na!Md wV
Q~>UL \Q%mľzfvck)6T*nGk`*֝=jS,,m$: `!9?Q@Q3 |`
qI.x0o9:ʺ3Flydi
a%PL7#вPCIe[!Yڀۡ$lǕdnI2ۭ)6b^,ϳ\A  -fa8Oю$-]&1gh	pbIӄ]q/*yG/q K{<XlT>٦/pSytVG='~;-Rv_
P@ 7,z;N
gnT7=cp	x_9ӑ4<hTDU&d%從/ql`>k|`o$yJ Rv	05y_s'!H^= !*sHMN{L+B=]{I9<Tɝ礊/|}JDyONNS[D28=В<jN;7myP(pyZ*].eƚm)֋|EuYeHuD$Ho\FtܖLCe±&_ps782c)>J ۢBOcW|PD`t&+
QZjAsn6t_]@,T9=L6?mقg,C(ejOdԻGUrIVjKrTS%TjRեNVd,>߯;Uj~S
Q'-;76'2x
Ed Ƽ!;;<Myk4t
D.zf:̤ɤit-aM3ɛDr^}V/
Q28
Pxetq38&o#nkK=4$,љUl p[&{p8¿\Sjh8O$wYꁝireG#`ߔ)l(P cjI#nCGS"YāGTݿlWIȮa?3t|Q.u[X>YD'o.?F`R!8Tٝ6vszgvYn+pKcwA
o.6	lc`v2[/>"k-LP$˄U#4S4E)Mfǔ8)婆8BGx8nS@MFFݥ<:KiHݣԡN
E K<T(q4bJS`I)UIڥep.K숖82=qf"t*PG^bo22=L;
To襮1:ژM&^߬YK:(Ae71(K~U@d۝4~⨍+P{Η!S3xw$e&##yJ:/OeSd{&qi-:LXљH&%h>c԰|1
a  PL
`KHT@efҶAsK#`[ˊsryyRnVi>` ]h¢c1`*5JGSȒD*n+>MC{8o7t]_GN!˼߳ m8/?6< KXɁ%Ih	`?HYK/9n2cf2	mhY(<:؏M-0jF{;y:ضH4=b0._'f+ɩxӦ~1rGrۃao;ߙlF|_^}Sೳ
]|ќ}tع?LN`oL~C?Er^?eo߃4(x49OSR$y̛2ûM"U
ߞ~ڏ"K~,1lGDh6DB7HiTT~L{vVf40Zт6mhߐ\3՝D7tsK
cP{V1NK^P+g5DȣB[orn;'YqW~64oџvm@/k4ThRb
MR&z]FitKCNDZokW ˱&ьyUpCk s!Ϳo
_F??R-ؠiT@
 ٩)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       ew.extension.jupyter.state.hidden":0,"memento/externalUriOpeners":1,"vscode.typescript-language-features":1,"workbench.view.extension.github-pull-requests.state.hidden   v>/Oms-python.debugpy{"PYTHON_GLOBAL_STORAGE_KEYS":[{"key":"PYTHON_EXTENSION_GLOBAL_STORAGE_KEYS","defaultValue":[]}]}$;Eworkbench.auxiliaryBar.emptyfalse1:cworkbench.auxiliaryBar.lastNonMaximizedSize300"85#mcp.lastSyncUserData{"ref":"0"}   <gextensionsAssignmentFilterProvider.copilotSkuno_access!6/'sync.lastSyncTime1771249155930'Z   SQLite format 3   @    1b      [   C                                                  1b ._
 \ \                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     ?     _        ? >C
u6om7d|:è_\ecR t@ZOi_ǒъH_/9ºkV/  kIF(evD jV~ r)6(ȂBSJB         v4.0.30319     l    6  #~  l6  ,  #Strings    Hc    #US e     #GUID   ,e    #Blob         
W+	   3     X   +      W  *                    .                                                4Y   HY Y 'Y Y l(Y 5Y Y J    Y Y +Y {Y -Y !% Y   Y 9 Y
 ' "   ^+     JQ   Q  JY "p,   (% Y  ^
` A   {  	{   &%Q  {  K{    kY Y V    Y Y Y A _ %Y Y 8	 Y " <
 
 4% Y q  S  4% *Y %  sY bY =  (      '+!    
9   9 9 a9 9 P9 m9 H9 Z9   |     l        "     /                       	
    WY       %  	  	Y  	  ,+ 
   .+= 
   %
 
   
 
   U!
  "  
=   
   =    
=   
   =     
    '  =   
   =    #
  =   
   =     T
      x     t
      x    ,
     
  	  H
    A
|     x    {
     
  %  2"
  :     I 
  Q   *
%  S  [
  S  %	  W  u    X  h  =  X& " &  & L#& #"M	VV1 Q( Z  % w% ,
 
 XH H  H H $ H H 8H H U H H j H  H H eH  H  H ?H H H  H #H pH H  H H H H  H 
H CH H _H -H tH BH H  H sH + H @ H H H q H H H  H H H iH H  H H  H FH H "H H H H 'H H H JH TH H lH H H fH 4H H {H H  H H zH H 2 H 1H PH G H \ H x H H bH  H H H  H  H H EH  H  H H H .H QH H &H mH H ;H 7H H IH H LH  H H H 9 H H WH H H >H N H c H  H H H  H ^H 
H  H H H  H H  H H *H H H  H H wH H )H H ~H H 0H 5H H H H XH H  H <H MH H H H  H [H H H SH 8} L
    L
! '! 7
! '! | !	 L
  q

 L
3 
! f
3 
3 
3 
! f
3 3 ! 3 >! f
3 K  t + v " 	V3 P        X        p              !              #c      1{ S!    	 m!     ?(
 t!     X
 !    F !     !       "    ++ 
"    <+ "     
 ."     6"     @"     } u"     N) "    (5 "    +, T#    
  ~#      #    e #      f$    3 x$     5 $    K+ $    ^+ $     $     $     ; 
%    OL %    L :%    ` K%    h \%    L t%    (L %    L %    L %    $L %    s&L &    "L .&    HL M&    
L l&    8)L &    L &     ,L &    
L &    
L '    #)L &'    	L E'    8L d'    fL '    
L '    )L '    L '    L '    L (    L =(    L \(    
L {(    L (    ,L (    	L (    BL (    	L )    L 5)    GL T)    ZL s)    "L )     L )    \L )    
L )    O)L *    CL -*    * L L*    8L k*    
L *    L *    L *    

L *    {
L +    rL %+    
L D+    L c+    ,L +    t
L +    ML +    #L +    L +    ,L ,    &L <,    L [,    L z,    L ,    	L ,    L ,    C	L ,    L -    })L 4-    $L S-    $L r-    )L -    )L -    dL -    L -    L 
.    'L ,.    #L K.    #L j.    ^#L .    E$L .    (L .    &L .    ;L /    !L $/    KL C/    a L b/    L /    C'L /    !L /    #L /    'L /    7L 0    zL ;0    G%L Z0    +	L y0    L 0    &L 0    (L 0    .L 0    Y	L 1    =L 31    )L R1    L q1    L 1    L 1    L 1    )L 1    ,#L 2    &L +2    L J2    	L i2    	L 2    OL 2    kL 2    -L 2    L 3    !L #3    $L B3    "L a3    ,L 3    L 3    t#L 3    O"L 3    !L 3    #L 4    ,L :4    L Y4    L x4    ~L 4    [&L 4    l	L 4    L 4    U$L 5    	L 25    	L Q5    L p5    L 5    L 5    L 5    L 5    L 6    #L *6    L I6    s(L h6    $L 6    L 6    	L 6    f)L 6    L 7    !L "7    L A7    h"L `7    %$L 7    ]L 7    &L 7    $L 7    "L 7    p$L 8    -'L 98    	L X8    #L w8    ?#L 8     S 8    8! 8    y# 9    Hy$ P9     & % 9     % % 8:    % p:    & :    (' :     $* M;    (5. \;     0 ;     <U4 =     '9 0>     
> ?     
? @    @ @     A @     A B     OD D     G F     OJ I     
M  I    +M XI     U'
N I     

O lJ    L

Q J     
S 4L    T 8L    UT LL    Y `L    (^ nL     (5_ }L     a L     b L     
f L     g L      g L    '
g M    ]
i LN     k XN     
k dN     
l N    n N    Un O    s "O    (x 0O     (5y ?O     { MO     | _O     
 uO      O     ! 4P    '
 |P    ]
 P      P     " Q     . R    4,A R    4,T R    !v R    ! XS    ! S    |% ,T    |% T    |% U     QU           ]
       '
 uU       }U    ($ U     ) U     1 U       U     
 U     U    U U     V    ( V     (5 -V      ;V      MV     
 cV      L       V    '
 W    ]
 W      W     ZV DX     a X    (l X    !l X     r X     { U              X     Y     Q       (5       
       
 Y    < Y     
 Z     <"
 Z    <5
 ([    <V
 [    ' @\     L]    <U]    U
^     
4_    (\_    _     c!
_     wuU      uU      _     
`     
1`    y@`    @&
x`    '`    .$a    r!
Va    B da    yta    %
a    .a    H!
 &b    o'4!8b     !b     "b     #b    +#^c    	*#fc    ( #mc    	*#|c    #c    #c     #c     0$c    (%c     %c     &U      ' d     j'      ? A,       ? A,       `,               x   [%     /  =      q+   x                     +   +   l,      l,                   Z,  (   }   % c   +   m      ?   N     
     Z        
     L  t      v
   v
            p
   !         p
   %         p
   !         p
   %   +  '     g   +  a
   F"   
     Z        
     L  t      +   +   m   +      ?   N                    F"   F"   ?   
     Z        
     L  t      +   +   m   +      ?   N                       "         "         U   Z,   U   Z,   U   
   
   Y   r
   U   
   
   Y   r
      U   
   
   Y   r
         U   
   
   Y   r
   U   
   
   Y   r
      U   
   
   Y   r
         U   (      ?   (                   F"   :!   F"   
     Z        
     L  t      +   +   m   +      ?   N                    "         "         F"   :!      ?   N        ?             t   +   m               +      +      O,   +   +      O,   +   
     L  t    
     L  t      
     Z      
     Z        +      +   +   O,   +   [   l   :!   .   P&   '      ~!   .   P&      ~!         t   -*   -*   F"   :!   C      (   +% 2 % % 6 %  %  	      !    P\ P\Q a 
Q  q  ~
9 } (  R 0,[ (a (h @t &~       $  &9      Q       [      ,  &  4  4  t4 f
4 @t< +$ ~
4 +, ~
$ ,, P\, D PD D  &    & ,   / 7@ cE ~
 yJ '$ P 2 ~
 A$ !1$ 
D 
9 a ,L kAuAI!A_I)AFIp
II IIQ+IIJ)A>A))p
1&A>))+1ATA	)G	aw 	ih2	A:	Q+D	))dI	)X	aFm	)G~	q   
 B y  T \  
`ivmii' i(&
i  Q ~
 !  iND ~
 0,D ,D ~
e(

N*
N?
N
$ ~
$ @tD @t

   d l *+
  t   t [  !  ( *	
1  9 A  I MY a i q y            
 p	  p	  p	    p	 ! p  3  p <p 3 %3' " w'  .  . #%. +.. 3T. ;]. B. C. K. S. S. S. S. [!. c<. kp. s. {. A. . SW. . C   C 
  c   c 
  c       
         
    &    
              B w#  # c  c       d           B w     BC BC w B w   B w*  # C  d  $*  d*  *  *  *  $	*  D	*  	*  	  	   A 
 	A #
   `    <          " $# # #  # #  $ $$  $w$$  %w % %`% B% B% B% B& ' (= ) @) ) ) ) )  *  * *                               +      Q1>t~
jV	



CPa
t





'
    
                 "  #  %  (    J    q  7-  b+  q  Sr  r  r  (r  r  r  $r  w&r  #r  Lr  
r  <)r  r  $,r  
r  
r  ')r  	r  <r  jr  
r  )r  r  r  r  r  r  
r  r  ,r  	r  Fr  	r  r  Kr  ^r  &r  r  `r  
r  S)r  Gr  . r  <r  
r  r  r  
r  
r  vr  r  r  ,r  x
r  Qr  #r  r  0r  *r  r  r  r  	r  r  G	r  r  )r  $r  $r  )r  )r  hr  r  
r  'r  #r  #r  b#r  d$r  8(r  &r  ?r  !r  Or  e r  r  G'r  !r  #r  'r  ;r  ~r  K%r  /	r  r  &r  !(r  2r  ]	r  Tr  )r  r  r  r  !r  )r  0#r  &r  r  	r  	r  Sr  or  1r  r  !r  $r  "r  ,r  r  x#r  S"r  !r  !#r  ,r  r  r  r  _&r  p	r  r  Y$r  	r  
r  r  r  r  r  r  r  'r  r  w(r  $r  r  	r  j)r  r  !r  r  l"r  )$r  ar  &r  $r  "r  t$r  1'r  	r  #r  C#r   
      (9    (  %    6*  s*  (              	         
    
  "   #   &   '   (   )   *   +   ,   - !  . #  / %  0 '  1 )  2 +  3 -  4 /  5 1  6 3  7 5  8 7  9 9  : ;  ; =  < ?  = A  > C  ? E  @ G  A I  B K  C M  D O  E Q  F S  G U  H W  I Y  J [  K ]  L _  M a  N c  O e  P g  Q i  R k  S m  T o  U q  V s  W u  X w  Y y  Z {  [ }  \   ]   ^   _   `   a   b   c   d   e   f   g   h   i   j   k   l   m   n   o   p   q   r   s   t   u   v   w   x   y   z   {   |   }   ~                                                                                                      	    
                    !  #  %  '  )  +  -  /  1  3  5  7  9  ;  =  ?  A  C  E  G  I  K  M  O  Q  S  U W Y  [ !] *_ La Nc Se% /% -% +% =% ?% )% AVuL d:M g                
              @                                &                ,                {                 P              u       	                  !   % $ + *     	    	5 [+ +} } }   }   }  }  }   } 	 }  }  +   + } }+#+ _u0000 _u1000 _u2000 _u3000 _uA000 _u0100 _u1100 _u2100 _u3100 _u1200 _u2200 _u3200 _u0300 _u2300 _u3300 _u0400 _u1400 _u2400 _u3400 _u0500 _u2500 _uA500 _u0600 _u2600 _u0700 _u1700 _u2700 _uA700 _u0800 _u1800 _u2800 _uA800 _u0900 _u1900 _u2900 _uA900 _uF900 _u0A00 _u1A00 _u2A00 _uAA00 _u0B00 _u1B00 _u2B00 _uAB00 _uFB00 _u0C00 _u1C00 _u2C00 _uAC00 _u0D00 _u1D00 _u2D00 _u0E00 _u1E00 _u2E00 _u4E00 _uFE00 _u0F00 _u1F00 _u2F00 _uFF00 _uFE10 _u1720 _uA720 _u1A20 _uFE20 _u3130 _u0530 _uA830 _uA930 _uAB30 _u2D30 _uFE30 _u3040 _u2440 _uA640 _u1740 _u0840 _uA840 _u2150 _u0250 _u0750 _u1950 _uFB50 _u1C50 _uFE50 _u2460 _u1760 _u0860 _uA960 _uAA60 _u2C60 _u2070 _u0370 _uAB70 _uFE70 _u0080 _u0180 _u1380 _u2580 _u1680 _u0780 _u1780 _uA880 _u0980 _u1980 _u2980 _uA980 _u0A80 _uAA80 _u0B80 _u1B80 _u0C80 _u1C80 _u2C80 _u0D80 _u1D80 _u2D80 _u0E80 _u2E80 _u2190 _u3190 _uA490 _u0590 _u1C90 _u10A0 _u20A0 _u30A0 _u31A0 _u13A0 _u25A0 _u16A0 _uA6A0 _u08A0 _u02B0 _uD7B0 _u18B0 _u1AB0 _u31C0 _u07C0 _u27C0 _u1BC0 _uABC0 _u1CC0 _u1DC0 _u4DC0 _u20D0 _uA4D0 _u1CD0 _uA8E0 _u19E0 _uA9E0 _uAAE0 _u2DE0 _u31F0 _u27F0 _u2FF0 _uFFF0 <EncodeUtf8>g__TryEncodeScalarAsHex|2_0 <EncodeUtf16>g__TryEncodeScalarAsHex|3_0 <EncodeUtf8>g__TryEncodeScalarAsHex|4_0 <EncodeUtf16>g__TryEncodeScalarAsHex|5_0 Vector64`1 Vector128`1 IEnumerable`1 ArrayPool`1 ReadOnlySpan`1 IEnumerator`1 Vector`1 StrongBox`1 AsUInt32 __StaticArrayInitTypeSize=8192 <i>5__2 Sse2 Log2 GetIndexOfFirstByteToEncodeSsse3 GetIndexOfFirstCharToEncodeSsse3 <GetAllowedCodePoints>d__14 GetIndexOfFirstByteToEncodeAdvSimd64 GetIndexOfFirstCharToEncodeAdvSimd64 Arm64 LoadVector64 AsUInt64 EncodeUtf16 DecodeFromUtf16 TryEncodeToUtf16 AsInt16 System.Runtime.Intrinsics.X86 8B30AFDCF07C4ABDFE0FAF65F79FC40A2E9AC497C42B1BA5C996BDFB3F6EC2F6 LoadVector128 LoadScalarVector128 FindFirstCharacterToEncodeUtf8 DecodeFromUtf8 TryEncodeToUtf8 TryEncodeUnicodeScalarUtf8 <Module> <PrivateImplementationDetails> get_ArabicExtendedA get_CyrillicExtendedA get_EthiopicExtendedA get_LatinExtendedA get_HangulJamoExtendedA get_MyanmarExtendedA get_CjkUnifiedIdeographsExtensionA get_MiscellaneousMathematicalSymbolsA get_ArabicPresentationFormsA get_SupplementalArrowsA get_CyrillicExtendedB get_LatinExtendedB get_HangulJamoExtendedB get_MyanmarExtendedB get_MiscellaneousMathematicalSymbolsB get_ArabicPresentationFormsB get_SupplementalArrowsB get_CyrillicExtendedC get_LatinExtendedC get_LatinExtendedD get_LatinExtendedE System.IO SR T value__ get_Kannada get_Lepcha get_Sinhala get_Thaana get_Hiragana get_Katakana get_Phagspa get_Saurashtra PopulatePreescapedData AsciiPreescapedData _asciiPreescapedData TryGetPreescapedData preescapedData pData data get_Tagbanwa get_Oriya FxResources.System.Text.Encodings.Web get_Syriac get_Mandaic get_Arabic get_Cyrillic get_Runic get_Ethiopic System.Collections.Generic ShiftRightArithmetic get_Glagolitic get_Coptic get_GreekandCoptic get_CurrentManagedThreadId <>l__initialThreadId Add get_EthiopicExtended get_DevanagariExtended get_GreekExtended get_GeorgianExtended get_BopomofoExtended get_UnifiedCanadianAboriginalSyllabicsExtended get_CombiningDiacriticalMarksExtended bytesConsumed charsConsumed ReadUnaligned WriteUnaligned SingletonMinimallyEscaped get_Shared get_IsHardwareAccelerated get_IsSupported IsScalarValueAllowed IsCharAllowed IsCodePointAllowed get_Buhid <Length>k__BackingField <FirstCodePoint>k__BackingField FixedElementField AdvSimd op_BitwiseAnd get_TaiLe Slice GetPinnableReference GetReference utf8Source source System.Text.Unicode WillEncode GetIndexOfFirstByteToEncode GetIndexOfFirstCharToEncode indexOfFirstCharToEncode FindFirstCharacterToEncode get_Cherokee System.Runtime.CompilerServices.Unsafe ToVector128Unsafe CreateScalarUnsafe ForbidRange UnicodeRange CreateRange AllowRange CreateEmptyRange range IEnumerable IDisposable RuntimeTypeHandle GetTypeFromHandle Shuffle Volatile GetArgumentName System.Runtime get_None _none Rune extraCharactersToEscape ValueType GetType EncodeUtf8Core EncodeCore ScalarEscaperBase get_Sundanese get_Javanese get_Buginese get_Balinese AddPairwise System.IDisposable.Dispose TryCreate IsHighSurrogate highSurrogate lowSurrogate PackUnsignedSaturate EditorBrowsableState <>1__state Write AssemblyMetadataAttribute EmbeddedAttribute CompilerGeneratedAttribute AttributeUsageAttribute NeutralResourcesLanguageAttribute IsByRefLikeAttribute DebuggableAttribute NullableAttribute EditorBrowsableAttribute AssemblyTitleAttribute IteratorStateMachineAttribute UnsafeValueTypeAttribute ObsoleteAttribute TargetFrameworkAttribute NotNullAttribute SupportedOSPlatformAttribute TargetPlatformAttribute InAttribute DebuggerHiddenAttribute ExtensionAttribute AssemblyFileVersionAttribute AssemblyInformationalVersionAttribute AssemblyDescriptionAttribute DoesNotReturnAttribute DefaultMemberAttribute FixedBufferAttribute NativeIntegerAttribute AssemblyDefaultAliasAttribute CompilationRelaxationsAttribute AssemblyProductAttribute AssemblyCopyrightAttribute SkipLocalsInitAttribute CLSCompliantAttribute NullableContextAttribute ParamArrayAttribute NullablePublicOnlyAttribute IsReadOnlyAttribute AssemblyCompanyAttribute RuntimeCompatibilityAttribute AsSByte AsByte get_NewTaiLue get_Value GetUtf16SurrogatePairFromAstralScalarValue GetUtf8RepresentationForScalarValue scalarValue value get_Size f Flag get_Rejang System.Threading System.Runtime.Versioning allowMinimalEscaping get_UnsafeRelaxedJsonEscaping allowMinimalJsonEscaping GetResourceString WritePartialString ToString GetString EncodeToNewString Casing casing get_BoxDrawing get_Tagalog TryGetSwitch get_Tifinagh Math get_Length set_Length requestedLength AreValidIndexAndLength get_Utf16SequenceLength spanRealLength bufferLength actualInputLength textLength length get_KayahLi get_Vai get_Thai get_Gurmukhi get_IsAscii get_OlChiki get_Bengali get_Devanagari get_SylotiNagri get_Gujarati get_Batak isFinalBlock get_MeeteiMayek MoveMask ShiftRightLogical get_MiscellaneousTechnical MemoryMarshal get_LatinExtendedAdditional CompareEqual System.ComponentModel get_Tamil get_All _all System.Text.Encodings.Web.dll _AssertThisNotNull ILLink.Substitutions.xml _arrayToReturnToPool IsControl get_Cham get_TaiTham get_Ogham get_Malayalam get_Item System System.Runtime.Intrinsics.Arm get_Bamum Enum Boolean ReadUInt32LittleEndian TryWriteUInt64LittleEndian GetDefinedBmpCodePointsBitmapLittleEndian IsLittleEndian get_Georgian get_Mongolian get_Armenian AppendSpan get_DefinedCharsBitmapSpan AsSpan span get_Tibetan get_Samaritan bytesWritten charsWritten numberOfCharactersWritten get_BasicLatin utf8Destination destination EscaperImplementation get_CjkSymbolsandPunctuation get_GeneralPunctuation get_SupplementalPunctuation System.Reflection get_OpticalCharacterRecognition PlatformNotSupportedException MissingManifestResourceException ArgumentOutOfRangeException ThrowArgumentNullException ArgumentException UnsafeRelaxedEscapingSingleton BasicLatinSingleton Return get_Kanbun get_NKo CopyTo get_Lao get_Bopomofo get_HangulJamo get_HangulCompatibilityJamo get_Hanunoo get_Zero _preescapedMap AsciiByteMap GetAllowedCodePointsBitmap _allowedCodePointsBitmap AllowedBmpCodePointsBitmap get_IsBmp allowedCodePointsBmp VectorTableLookup TryLookup Clear ForbidChar InsertAsciiChar get_ReplacementChar ThrowArgumentException_MaxOutputCharsPerInputChar get_TextEncoderDoesNotImplementMaxOutputCharsPerInputChar AllowChar TryEncodeUnicodeScalar unicodeScalar ToScalar scalar get_Myanmar ValueStringBuilder DefaultHtmlEncoder DefaultUrlEncoder _innerEncoder DefaultJavaScriptEncoder OptimizedInboxTextEncoder <Data>e__FixedBuffer <Bitmap>e__FixedBuffer <Buffer>e__FixedBuffer <AsBytes>e__FixedBuffer utf16ScratchBuffer initialBuffer ToBytesBuffer ToCharsBuffer buffer get_ResourceManager s_resourceManager other get_Khmer _scalarEscaper ThrowHelper ExtractNarrowingSaturateUnsignedUpper ToCharUpper ForbidCharacter lastCharacter firstCharacter get_MaxOutputCharactersPerInputCharacter AllowCharacter character TextWriter writer BitConverter HexConverter ExtractNarrowingSaturateUnsignedLower IEnumerator System.Collections.Generic.IEnumerable<System.Int32>.GetEnumerator System.Collections.IEnumerable.GetEnumerator .ctor .cctor get_AsVector IntPtr As get_UnifiedCanadianAboriginalSyllabics System.Numerics get_EnclosedAlphanumerics System.Runtime.Intrinsics System.Diagnostics System.Runtime.InteropServices System.Runtime.CompilerServices System.Resources FxResources.System.Text.Encodings.Web.SR.resources DebuggingModes allowedRanges ForbidRanges UnicodeRanges ValidateRanges AllowRanges ranges get_CjkStrokes get_YiSyllables get_HangulSyllables get_GeometricShapes get_ControlPictures TryWriteBytes lengthInBytes AsBytes BinaryPrimitives NullableFlags TransformFlags TextEncoderSettings settings get_CjkUnifiedIdeographs get_CjkCompatibilityIdeographs get_EnclosedCjkLettersandMonths <>4__this System.Diagnostics.CodeAnalysis Microsoft.CodeAnalysis get_CombiningHalfMarks get_CombiningDiacriticalMarks get_YiRadicals get_KangxiRadicals get_Specials IncludesInternals get_LetterlikeSymbols get_YijingHexagramSymbols get_KhmerSymbols get_CombiningDiacriticalMarksforSymbols get_MiscellaneousSymbols get_CurrencySymbols get_HalfwidthandFullwidthForms get_VerticalForms get_AlphabeticPresentationForms get_NumberForms get_CommonIndicNumberForms get_CjkCompatibilityForms get_IpaExtensions get_VedicExtensions get_PhoneticExtensions get_KatakanaPhoneticExtensions get_MeeteiMayekExtensions TextWriterExtensions MemoryExtensions BitOperations System.Collections get_BraillePatterns additionalCapacityBeyondPos _pos TryWriteChars lengthInChars _chars System.Buffers UnicodeHelpers ForbidUndefinedCharacters ForbidCharacters forbidHtmlSensitiveCharacters ForbidHtmlCharacters get_IdeographicDescriptionCharacters AllowCharacters characters get_ModifierToneLetters get_SpacingModifierLetters get_MathematicalOperators get_SupplementalMathematicalOperators System.Numerics.Vectors get_VariationSelectors ReverseEndianness get_Dingbats AttributeTargets get_SmallFormVariants get_BlockElements PopulateAllowedCodePoints GetAllowedCodePoints AllowedAsciiCodePoints _allowedAsciiCodePoints _allowedBmpCodePoints AllowCodePoints codePoints get_SuperscriptsandSubscripts OperationStatus get_Arrows get_MiscellaneousSymbolsandArrows UsingResourceKeys s_usingResourceKeys Concat Object get_TaiViet System.Collections.IEnumerator.Reset requestedOffset _GetIndexAndOffset AddByteOffset offset op_Implicit get_Default Rent get_Latin1Supplement get_SyriacSupplement get_ArabicSupplement get_CyrillicSupplement get_EthiopicSupplement get_CherokeeSupplement get_SundaneseSupplement get_GeorgianSupplement get_CombiningDiacriticalMarksSupplement get_CjkRadicalsSupplement get_PhoneticExtensionsSupplement Environment ExceptionArgument argument System.Collections.Generic.IEnumerator<System.Int32>.Current System.Collections.IEnumerator.Current System.Collections.Generic.IEnumerator<System.Int32>.get_Current System.Collections.IEnumerator.get_Current <>2__current IsAllowedAsciiCodePoint IsAsciiCodePoint IsBmpCodePoint get_FirstCodePoint set_FirstCodePoint firstCodePoint codePoint get_Count TrailingZeroCount characterCount count last CompareTest first output MoveNext System.Text utf8Text AppContext get_Limbu get_Telugu get_Lisu get_Hebrew Grow Max IsValidIndex startingIndex startIndex index resourceKey key System.Buffers.Binary System.Memory op_Equality get_CjkCompatibility UnicodeUtility SpanUtility get_IsEmpty  kT e x t E n c o d e r D o e s N o t I m p l e m e n t M a x O u t p u t C h a r s P e r I n p u t C h a r  MS y s t e m . R e s o u r c e s . U s e S y s t e m R e s o u r c e K e y s  f i r s t C o d e P o i n t  
l e n g t h  l a s t C h a r a c t e r  s e t t i n g s  v a l u e  
o u t p u t  s t a r t I n d e x  c h a r a c t e r C o u n t  o t h e r  a l l o w e d R a n g e s  c h a r a c t e r s  c o d e P o i n t s  r a n g e  
r a n g e s     ǛXdNMK^V  $          $  RSA1     KxT4; ;9 t6_qU)ϻ;"lj7
LHkn"Y8Gnq̒a< ec0ɠAU$)A3:پ+8?_
:{-Q       & k   T
AllowMultiple T	Inherited & L   T
AllowMultiple T	Inherited  &     T
AllowMultiple T	Inherited 		!   !$    ! !$ 	$    )     15 1  )      )  	! !   !
 !    A !   A !
!E  E  			   !  	 !   (  X RTypes with embedded references are not supported in this version of your compiler.   	 Item    A 	
 	A  A  8   8  8	  
8 88	 88 8    D 1g ^System.Byte, System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a         iLi `System.UInt32, System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a      	E	A!	u	!	u	 A u u u u  	 A 	 !  	A  	 	T\Hx	AA	   xHA g bFindFirstCharacterToEncode has been deprecated. It should only be used by the TextEncoder adapter.  !	 c ^TryEncodeUnicodeScalar has been deprecated. It should only be used by the TextEncoder adapter.  
i
 i i   i
 ! A!	i
 Ai
 ! A!Ei    AE A i  1 11gT	   	  
  	  		 
		 
 

 		  
    mT	 
  
   
    	 		    
 
Xg ^System.Byte, System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a       	 H( `i `System.UInt64, System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a     !!i	 i Hx 	dPxH!  h
 i!
 i!lPxH!xH p@ i! i!A A ! !  !   !	
  
 !		 !	 
 !
 !	
   ! 
 !   | | |8 |A
 	!
 	!    8        E 	!i	 Ai 	!! AA0!! 
 A 
 AA	 A 
 A	 !   
 AA!!i	AAi! A A 	i  8 		      	  8  N ISystem.Text.Encodings.Web.TextEncoderSettings+<GetAllowedCodePoints>d__14  HH   H      (       8  A	         TWrapNonExceptionThrows )      5 .NETCoreApp,Version=v6.0 TFrameworkDisplayName  System.Text.Encodings.Web  
 en-US    .NETFrameworkAssembly    ServiceableTrue   PreferInboxTrue   IsTrimmableTrue   Microsoft Corporation  3 .© Microsoft Corporation. All rights reserved.   Internal implementation package not meant for direct consumption. Please do not reference directly. 
Provides types for encoding and escaping strings for use in JavaScript, HyperText Markup Language (HTML), and uniform resource locators (URL).

Commonly Used Types:
System.Text.Encodings.Web.HtmlEncoder
System.Text.Encodings.Web.UrlEncoder
System.Text.Encodings.Web.JavaScriptEncoder   6.0.2223.42425  4 /6.0.22+4bb6dc195c0a3bc4c7e24ff54a8925b98db4fecd   Microsoft® .NET  4 
RepositoryUrl!https://github.com/dotnet/runtime   
Browser1.0   Browser         9Ҩ MP   ~   D  D                           RSDSǛXdNMK^V   /__w/1/s/artifacts/obj/System.Text.Encodings.Web/net6.0-Browser-Release/System.Text.Encodings.Web.pdb                                                          _CorDllMain mscoree.dll     %  @                                                                                                                                                                                                                                                                   0                     H   X            4   V S _ V E R S I O N _ I N F O                 ?                         D    V a r F i l e I n f o     $    T r a n s l a t i o n       |   S t r i n g F i l e I n f o   X   0 0 0 0 0 4 b 0    C o m m e n t s   I n t e r n a l   i m p l e m e n t a t i o n   p a c k a g e   n o t   m e a n t   f o r   d i r e c t   c o n s u m p t i o n .   P l e a s e   d o   n o t   r e f e r e n c e   d i r e c t l y .   
 P r o v i d e s   t y p e s   f o r   e n c o d i n g   a n d   e s c a p i n g   s t r i n g s   f o r   u s e   i n   J a v a S c r i p t ,   H y p e r T e x t   M a r k u p   L a n g u a g e   ( H T M L ) ,   a n d   u n i f o r m   r e s o u r c e   l o c a t o r s   ( U R L ) . 
 
 C o m m o n l y   U s e d   T y p e s : 
 S y s t e m . T e x t . E n c o d i n g s . W e b . H t m l E n c o d e r 
 S y s t e m . T e x t . E n c o d i n g s . W e b . U r l E n c o d e r 
 S y s t e m . T e x t . E n c o d i n g s . W e b . J a v a S c r i p t E n c o d e r     L   C o m p a n y N a m e     M i c r o s o f t   C o r p o r a t i o n   \   F i l e D e s c r i p t i o n     S y s t e m . T e x t . E n c o d i n g s . W e b   >   F i l e V e r s i o n     6 . 0 . 2 2 2 3 . 4 2 4 2 5     \   I n t e r n a l N a m e   S y s t e m . T e x t . E n c o d i n g s . W e b . d l l    .  L e g a l C o p y r i g h t      M i c r o s o f t   C o r p o r a t i o n .   A l l   r i g h t s   r e s e r v e d .   d   O r i g i n a l F i l e n a m e   S y s t e m . T e x t . E n c o d i n g s . W e b . d l l   @   P r o d u c t N a m e     M i c r o s o f t    . N E T    0  P r o d u c t V e r s i o n   6 . 0 . 2 2 + 4 b b 6 d c 1 9 5 c 0 a 3 b c 4 c 7 e 2 4 f f 5 4 a 8 9 2 5 b 9 8 d b 4 f e c d   8   A s s e m b l y   V e r s i o n   6 . 0 . 0 . 0                                                                                                                                                      1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      (    0(	*H
(0(10
	`He 0\
+7N0L0
+70	  010
	`He  pE0peJFpbZn*ˆy
0c0K3  |Y    |0
	*H
 0~10	UUS10U
Washington10URedmond10U
Microsoft Corporation1(0&UMicrosoft Code Signing PCA 20110
230511190332Z
240508190332Z0c10	UUS10U
Washington10URedmond10U
Microsoft Corporation1
0U.NET00
	*H
  0 |kH.E= e!1¨#Lyp>wTDbL
  ڄ?`$ ܒboTcZV\x%NC
*ZiP-ln/M;qf`R.?e$5fBZt	&sXM ruԷ8x"itv ."hS@ۘh)I.\QR鱶H72)&%dRwc*4LTj\[JE~6LJ{BsGtr~xJDF	|a<z0ZB_6_kaD@VT~"THk{Fk	=;]N:J{)-f,Wa=pON"pܜ[/ s0o0U%0
+7L+0URjaGHF\C"wy 0>I0EU>0<:0810UMicrosoft Corporation10U
464223+5010150U#0HndPӂ77"mu0TUM0K0IGEChttp://www.microsoft.com/pkiops/crl/MicCodSigPCA2011_2011-07-08.crl0a+U0S0Q+0Ehttp://www.microsoft.com/pkiops/certs/MicCodSigPCA2011_2011-07-08.crt0U0 0
	*H
  HFkd8{M٘FTlZUUI_
&SZX	n|z{>oZS^qSe*'\Ø tɉCJ/n[`Y&$T<!#v.$!%r{%ΒW zLC
6aGCvG"#~=n~Ìn[z5W,.+e{`mLjH裖I>cc }5J/׷1Lo綔s-+ј܇1jW\ c!)G@D#O<;i"R?f`+KwZdϨ()XEA8	gW12Ǆ}F΂7l2~UC
*xF3
rtp@.5zڽ	aeQ3c gqSmn4e96Q8q<xk'uGo4M)K1LALJ36 M0z0b
a     0
	*H
 010	UUS10U
Washington10URedmond10U
Microsoft Corporation1200U)Microsoft Root Certificate Authority 20110
110708205909Z
260708210909Z0~10	UUS10U
Washington10URedmond10U
Microsoft Corporation1(0&UMicrosoft Code Signing PCA 20110"0
	*H
  0
 r.nM4X!B*kZP8I7k891BӉ
yd~`$lIh^ߛS
,ï٩+z	זY`fv2R&/PϳD].uhm:\F8dnd5xc-@
\yI)
w=+ZMV(zr:I^C{{>]O/Y;-3Xŷ>2ӳ=8~ҝ,NV!5%9ndS#_R!t%_('(8%9J6|#faj(I_%]!K>RĵW?$z[/#p]QFw Ἤ_UK9#ID|Erzr$߿FWۃMI0[ݰf{KI)(}gbx_/W\w((m(,@7OD	LԥC/tnx X,]`>O3ڰޞNFlֈ 00	+7 0UHndPӂ77"mu0	+7
 S u b C A0U0U00U#0r-:1CN1#40ZUS0Q0OMKIhttp://crl.microsoft.com/pki/crl/products/MicRooCerAut2011_2011_03_22.crl0^+R0P0N+0Bhttp://www.microsoft.com/pki/certs/MicRooCerAut2011_2011_03_22.crt0U 00	+7.00?+3http://www.microsoft.com/pkiops/docs/primarycps.htm0@+042  L e g a l _ p o l i c y _ s t a t e m e n t . 0
	*H
  g򆥘Ty.tg"cB}ye_.,>rm?
;Gic"5]e}FPU|K<BRаk>.)=Եw9dD=xz#}t&F*hh.*?*kXIc	inZF+;н5n%'(M=2-[I9

ƭQEX82q'',: iYhnot@*(>?MfMo}Ev@*e]Rbc6'ͥ]cA[>wuaaA(e/\ ֡FY	KH )ű?Ew5Ңzz"HG#kKux7yKֳ#hui;0Lumc=ѝVNOW"x2!zA
?̤]WqVNHEBɛv[
HkymmjRs
zP1"000~10	UUS10U
Washington10URedmond10U
Microsoft Corporation1(0&UMicrosoft Code Signing PCA 20113  |Y    |0
	`He 0	*H
	1
+70
+710
+70/	*H
	1" 7b\wN%8i9v֟Q.0B
+71402 M i c r o s o f thttp://www.microsoft.com0
	*H
 g*c<ňszrQ(]ni4:n#wFeZE$Ӈ^<#h+( 4i'
$Zj?#1QvDg0z:y
xH㚧Ax䍩ÄN/;
Z d^CrRh,CgX/
ak*WGBL0=?.9dUH	@<OfvI%Njs$J
